### PR TITLE
Feature/311 display project icon

### PR DESCRIPTION
**Pull request by Ruby77**
## Description

Display project icon if an icon is uploaded, otherwise show the default placeholder.

## Type of change

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] This change requires a documentation update

## Checklist

-   [x] My code follows the style guidelines of this project
-   [x] I have performed a self-review of my own code
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have made corresponding changes to the documentation
-   [ ] I have added tests that prove my fix is effective or that my feature works
-   [x] I updated the changelog with an end-user readable description
-   [x] I assigned this pull request to the correct project board to update the sprint board

## Steps to Test or Reproduce

1. Add a project with a project icon.
2. Go to the project overview and check if the project icon is shown.
3. Click on the project to navigate to the details page and also check here if the project icon is shown.
4. Highlight this project, navigate to the homepage. Refresh this page until the project is shown and check if the project icon is visible. 

5. (This is currently not working, but if you press 'Embed this' on the details page, the project icon will also show. This problem is already available in the project icon branch I'm building on.)

## Link to issue

Closes: #311

### CODE REVIEW

File: src/app/modules/highlight/top-highlight-cards/top-highlight-cards.component.ts [hunk 1]
```diff
@@ -17,10 +17,13 @@
 
 import { HttpErrorResponse } from '@angular/common/http';
 import { Component, OnInit } from '@angular/core';
+import { DomSanitizer, SafeUrl } from '@angular/platform-browser';
 import { Router } from '@angular/router';
 import { finalize } from 'rxjs/internal/operators/finalize';
 import { Highlight } from 'src/app/models/domain/hightlight';
+import { Project } from 'src/app/models/domain/project';
 import { HighlightService } from 'src/app/services/highlight.service';
+import { RESOURCE_CONFIG } from '../../../config/resource-config';
```
Comment (waltersajtos) [inline src/app/modules/highlight/top-highlight-cards/top-highlight-cards.component.ts:26]: Make sure to use absolute paths

File: src/app/modules/project/embed/embed.component.ts [hunk 2]
```diff
@@ -18,6 +18,8 @@ import { EmbedService } from 'src/app/services/embed.service';
 import { Component, OnInit } from '@angular/core';
 import { ActivatedRoute } from '@angular/router';
 import { Project } from 'src/app/models/domain/project';
+import { SafeUrl, DomSanitizer } from '@angular/platform-browser';
+import { RESOURCE_CONFIG } from '../../../config/resource-config';
```
Comment (waltersajtos) [inline src/app/modules/project/embed/embed.component.ts:22]: Same here.

File: src/app/modules/project/overview/overview.component.ts [hunk 3]
```diff
@@ -281,6 +283,18 @@ export class OverviewComponent implements OnInit {
     this.onInternalQueryChange();
   }
 
+    /**
+   * Method to get the url of the icon of the project. This urls can be the local
+   * image for a default or a specified icon stored on the server.
+   */
+  public getIconUrl(id: number): SafeUrl {
```
Comment (waltersajtos) [inline src/app/modules/project/overview/overview.component.ts:290]: I see this function multiple times which isn't really dry.

**Review by waltersajtos — CHANGES_REQUESTED** [review]

**Review by waltersajtos — APPROVED** [review]
Changes were implemented properly, looks good!

**Review by liekevdvoort — APPROVED** [review]

**Pull request merged**